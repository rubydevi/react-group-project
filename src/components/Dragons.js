import Container from 'react-bootstrap/Container';

const DragonsComponents = () => (
  <Container><h1>Dragons Page</h1></Container>
);

export default DragonsComponents;
