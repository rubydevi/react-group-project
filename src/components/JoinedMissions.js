import { ListGroup } from 'react-bootstrap';
import { useSelector } from 'react-redux/es/hooks/useSelector';

const JoinedMissions = () => {
  const missions = useSelector((state) => state.missions.missions);
  const joinmission = missions.filter((mission) => mission.reserved === true);

  return (
    <>
      {joinmission.length > 0 ? (
        <ListGroup>
          {joinmission.map((mission) => (
            <ListGroup.Item key={mission.mission_id}>{mission.mission_name}</ListGroup.Item>
          ))}
        </ListGroup>
      ) : (<div>You have no Reserved Missions</div>)}
    </>
  );
};

export default JoinedMissions;
