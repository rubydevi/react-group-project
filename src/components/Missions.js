import { useEffect } from 'react';
import Container from 'react-bootstrap/Container';
import Button from 'react-bootstrap/Button';
import { useDispatch, useSelector } from 'react-redux';
import Table from 'react-bootstrap/Table';
import PropTypes from 'prop-types';
import { getMission, joinMission, leaveMission } from '../Redux/Missions/missions';

const MissionsComponent = () => {
  const missions = useSelector((state) => state.missions.missions);
  const dispatch = useDispatch();
  useEffect(() => {
    if (missions.length === 0) {
      dispatch(getMission());
    }
  }, [dispatch, missions.length]);

  const join = (id) => {
    dispatch(joinMission(id));
  };

  const leave = (id) => {
    dispatch(leaveMission(id));
  };

  return (
    <Container>
      <TableComponent
        missions={missions}
        leave={leave}
        join={join}
      />
    </Container>
  );
};

export const TableComponent = ({ missions, leave, join }) => (
  <Table striped bordered hover>
    <thead>
      <tr>
        <th>Mission</th>
        <th>Description</th>
        <th>Status</th>
        <th>.</th>
      </tr>
    </thead>
    <tbody>
      {missions.map((mission) => (
        <tr key={mission.mission_id}>
          <td>{mission.mission_name}</td>
          <td>{mission.description}</td>
          <td>
            {mission.reserved
              ? (<Button variant="success">Active Member</Button>)
              : (<Button variant="secondary">NOT A MEMBER</Button>)}
          </td>
          <td>
            {mission.reserved
              ? (<Button variant="outline-danger" onClick={() => leave(mission.mission_id)}>Leave Mission</Button>)
              : (<Button variant="outline-secondary" onClick={() => join(mission.mission_id)}>Join Mission</Button>)}
          </td>
        </tr>
      ))}
    </tbody>
  </Table>
);

TableComponent.propTypes = {
  missions: PropTypes.array.isRequired,
  leave: PropTypes.func.isRequired,
  join: PropTypes.func.isRequired,
};

export default MissionsComponent;
