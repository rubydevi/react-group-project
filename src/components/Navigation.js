import { NavLink } from 'react-router-dom';
import logo from '../images/planet.png';

const Navigation = () => (
  <div>
    <ul className="menu">
      <li className="logo">
        <img src={logo} alt="logo" />
        <h3>Space Travelers&apos; Hub</h3>
      </li>
      <li>
        <ul className="nav">
          <li>Rockets</li>
          <li>Missions</li>
          <li>Dragons</li>
          <li className="divider">|</li>
          <li>
            <NavLink className="nav-item" to="/profile" activeclassname="active">My Profile</NavLink>
          </li>
        </ul>
      </li>

    </ul>
  </div>
);

export default Navigation;
