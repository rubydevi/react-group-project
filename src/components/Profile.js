import Container from 'react-bootstrap/Container';

const ProfileComponent = () => (
  <Container><h1>Profile Page</h1></Container>
);

export default ProfileComponent;
