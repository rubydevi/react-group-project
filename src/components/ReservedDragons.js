import React from 'react';
import { useSelector } from 'react-redux';
import ListGroup from 'react-bootstrap/ListGroup';
import { Container } from 'react-bootstrap';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Badge from 'react-bootstrap/Badge';

const ReservedDragons = () => {
  const { dragons } = useSelector((state) => state.dragons);
  const reservedDragons = dragons.filter((dragon) => dragon.reserved === true);

  return (
    <Container className="mt-4">
      <Row>
        {reservedDragons.length > 0 ? (
          <Col>
            <ListGroup>
              {
            reservedDragons.map((dragon) => (
              <ListGroup.Item key={dragon.id}>{dragon.name}</ListGroup.Item>
            ))
              }
            </ListGroup>
          </Col>
        ) : (
          <Col>
            <h5>
              Dragons
              {' '}
              <Badge bg="info">No dragon reserved</Badge>
            </h5>
          </Col>
        )}

      </Row>
    </Container>

  );
};

export default ReservedDragons;
