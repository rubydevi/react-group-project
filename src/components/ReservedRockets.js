import React from 'react';
import { useSelector } from 'react-redux';

const ReservedRockets = () => {
  const rockets = useSelector((state) => state.rockets.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);

  if (reservedRockets.length === 0) {
    return <div>You have no reserved rockets.</div>;
  }

  return (
    <ul className="reserved-rockets">
      {reservedRockets.map((rocket) => (
        <li className="reserved-rocket" key={rocket.id}>{rocket.name}</li>
      ))}
    </ul>
  );
};

export default ReservedRockets;
