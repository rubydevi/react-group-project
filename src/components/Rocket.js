import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { reserveRocket } from '../Redux/Rockets/rocketsSlice';
import '../styles/Rocket.css';

const Rocket = ({ rocket }) => {
  const dispatch = useDispatch();

  const handleReserveRocket = () => {
    dispatch(reserveRocket(rocket.id));
  };

  return (
    <li className="rocket" key={rocket.id}>
      {rocket.flickr_images.length > 0 && (
      <img className="rocket-image" src={rocket.flickr_images[0]} alt={`Rocket ${rocket.name}`} />
      )}
      <div className="rocket-details">
        <h3 className="rocket-title">{rocket.name}</h3>
        {/* Show badge when reserved */}
        {rocket.reserved && <span className="reserved-badge">Reserved</span>}
        <p className="rocket-description">
          {rocket.description}
        </p>
        <button className="reserve-button" type="submit" onClick={handleReserveRocket}>
          Reserve Rocket
        </button>
      </div>
    </li>
  );
};

Rocket.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    flickr_images: PropTypes.arrayOf(PropTypes.string).isRequired,
    reserved: PropTypes.bool.isRequired,
  }).isRequired,
};

export default Rocket;
