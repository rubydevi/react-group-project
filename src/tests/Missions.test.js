import renderer from 'react-test-renderer';
import { screen, render } from '@testing-library/react';
import { TableComponent } from '../components/Missions';

const store = ({
  missions: {
    missions: [
      {
        mission_name: 'Thaicom',
        mission_id: '9D1B7E4',
        wikipedia: 'https://en.wikipedia.org/wiki/Thaicom',
        website: 'http://www.thaicom.net/en/satellites/overview',
        twitter: 'https://twitter.com/thaicomplc',
        description: 'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
        reserved: false,
      },
      {
        mission_name: 'Thaicom',
        mission_id: '9D1B7E0',
        wikipedia: 'https://en.wikipedia.org/wiki/Thaicom',
        website: 'http://www.thaicom.net/en/satellites/overview',
        twitter: 'https://twitter.com/thaicomplc',
        description: 'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
        reserved: false,
      },
      {
        mission_name: 'Thaicom',
        mission_id: '9D1B7E5',
        wikipedia: 'https://en.wikipedia.org/wiki/Thaicom',
        website: 'http://www.thaicom.net/en/satellites/overview',
        twitter: 'https://twitter.com/thaicomplc',
        description: 'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
        reserved: false,
      },
    ],
    status: 'success',
  },
});

describe('Testing Mission Components', () => {
  it('should render the mission component', () => {
    const leave = () => {};
    const join = () => {};
    const missions = renderer.create(
      <TableComponent
        missions={store.missions.missions}
        leave={leave}
        join={join}
      />,
    ).toJSON();
    expect(missions).toMatchSnapshot();
  });

  it('should render the mission component', () => {
    const leave = () => {};
    const join = () => {};
    render(
      <TableComponent
        missions={store.missions.missions}
        leave={leave}
        join={join}
      />,
    );
    const table = screen.getAllByRole('button');
    expect(table).toHaveLength(6);
  });
});
