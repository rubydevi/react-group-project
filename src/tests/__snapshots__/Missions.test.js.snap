// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Mission Components should render the mission component 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th>
        .
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Thaicom
      </td>
      <td>
        Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
      </td>
      <td>
        <button
          className="btn btn-secondary"
          disabled={false}
          type="button"
        >
          NOT A MEMBER
        </button>
      </td>
      <td>
        <button
          className="btn btn-outline-secondary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
    <tr>
      <td>
        Thaicom
      </td>
      <td>
        Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
      </td>
      <td>
        <button
          className="btn btn-secondary"
          disabled={false}
          type="button"
        >
          NOT A MEMBER
        </button>
      </td>
      <td>
        <button
          className="btn btn-outline-secondary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
    <tr>
      <td>
        Thaicom
      </td>
      <td>
        Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
      </td>
      <td>
        <button
          className="btn btn-secondary"
          disabled={false}
          type="button"
        >
          NOT A MEMBER
        </button>
      </td>
      <td>
        <button
          className="btn btn-outline-secondary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
